### PR TITLE
RELATED: SD-1233 add API for get all workspace datasets

### PR DESCRIPTION
**Pull request by lvkhanh**
- Add new API for   (`/gdc/md/${projectId}/query/datasets`) in workspace datasets service that return IMetadataObject[] items. That API needed on AD/KD to check workspace has existed datasets or not to route Data tab for navigation of freemium users

---

Supported PR commands:

| Command         | Description            |
| --------------- | ---------------------- |
| `ok to test`    | Re-run standard checks |
| `extended test` | BackstopJS tests       |

---

# PR Checklist

-   [x] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [x] `check` passes
-   [x] `check-extended` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by lvkhanh** [comment]
check-extended

**Comment by lvkhanh** [comment]
extended test

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/1698/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/3277/ : SUCCESS in 24m 12s

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/1699/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/530/ : SUCCESS in 30m 10s

**Comment by lvkhanh** [comment]
extended test

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/1704/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/3284/ : SUCCESS in 18m 38s

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/1705/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/531/ : SUCCESS in 24m 53s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/1706/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/3285/ : SUCCESS in 13m 54s

File: libs/sdk-backend-spi/src/workspace/ldm/datasets.ts [hunk 1]
```diff
@@ -1,17 +1,25 @@
...
+    getCsvDatasets(): Promise<ICsvDataset[]>;
+
     /**
      * Receive all workspace datasets
      *
      * @returns promise of workspace datasets
      */
-    getDatasets(): Promise<IDataset[]>;
+    getAllDatasets(): Promise<IMetadataObject[]>;
 }
```
Comment (lupko) [inline libs/sdk-backend-spi/src/workspace/ldm/datasets.ts:25]: we cannot do these breaking changes

File: libs/sdk-backend-spi/src/workspace/fromModel/ldm/datasets/index.ts [hunk 2]
```diff
@@ -67,7 +67,7 @@ export interface IDataHeader {
  *
  * @public
  */
-export interface IDataset {
+export interface ICsvDataset {
     dataset: {
         name: string;
         dataHeader: IDataHeader;
```
Comment (lupko) [inline libs/sdk-backend-spi/src/workspace/fromModel/ldm/datasets/index.ts:73]: also o breaking change

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/1741/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/3326/ : SUCCESS in 14m 37s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/1742/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/3327/ : SUCCESS in 14m 05s

**Comment by lvkhanh** [comment]
extended test

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/1767/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/3363/ : SUCCESS in 24m 11s

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/1770/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/540/ : SUCCESS in 14m 06s

File: libs/sdk-backend-spi/src/workspace/ldm/datasets.ts [hunk 3]
```diff
@@ -9,9 +10,16 @@ import { IDataset } from "../fromModel/ldm/datasets";
...
+     * @returns promise of workspace csv datasets
      */
     getDatasets(): Promise<IDataset[]>;
+
+    /**
+     * Receive all workspace all datasets
+     *
+     * @returns promise of workspace datasets
+     */
+    getAllDatasets(): Promise<IMetadataObject[]>;
```
Comment (lupko) [inline libs/sdk-backend-spi/src/workspace/ldm/datasets.ts:24]: I suggest to rename to getAllDatasetsMeta() to indicate that the function returns just the metadata and not full information about the data set.

File: libs/sdk-backend-spi/src/workspace/ldm/datasets.ts [hunk 4]
```diff
@@ -9,9 +10,16 @@ import { IDataset } from "../fromModel/ldm/datasets";
...
-     * Receive all workspace datasets
+     * Receive all workspace csv datasets
      *
-     * @returns promise of workspace datasets
+     * @returns promise of workspace csv datasets
      */
     getDatasets(): Promise<IDataset[]>;
+
+    /**
+     * Receive all workspace all datasets
```
Comment (lupko) [inline libs/sdk-backend-spi/src/workspace/ldm/datasets.ts:20]: please change: Receive all workspace dataset metadata.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/1788/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/3390/ : SUCCESS in 14m 06s

File: libs/sdk-backend-spi/src/workspace/ldm/datasets.ts (hunk 4 above)
Comment (lvkhanh) [reply libs/sdk-backend-spi/src/workspace/ldm/datasets.ts]: updated

File: libs/sdk-backend-spi/src/workspace/ldm/datasets.ts (hunk 3 above)
Comment (lvkhanh) [reply libs/sdk-backend-spi/src/workspace/ldm/datasets.ts]: updated

**Comment by lvkhanh** [comment]
I fixed your comments, @lupko. Thanks

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/1789/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/3391/ : SUCCESS in 14m 09s

**Review by lupko — APPROVED** [review]
LGTM

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/579/ : SUCCESS in 49s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/840/ : SUCCESS in 16s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/3418/ : SUCCESS in 25m 21s

**Pull request merged**